functions/routes/categoryRoutes.js

```diff
@@ -0,0 +1,93 @@
+const router = require('express')
+                    .Router();
+const categoriesRef = require('firebase-admin')
+                            .firestore()
+                            .collection('category');
+
+                            
+router
+    .route('/')
+    .get(getAllCategories)
+    .post(addNewCategory)
+    .put(updateCategory)
+    .delete(deleteCategory);
+
+
+function getAllCategories(req, res) {
+    categoriesRef
+        .get()
+        .then( (categories) => {
+            return res.status(200).json({
+                success: true,
+                message: 'List of All Categories Received Successfully',
+                data: {
+                    categories: categories.docs.map(doc => doc.data())
+                }
+            })
+        })
+        .catch( (error) => {
+            return res.status(500).json({
+                success: false,
+                message: error
+            })
+        });
+}
+
+function addNewCategory(req,res) {
+    categoriesRef
+        .doc(req.body.id)
```

Comment:
check if req.body.id is null or undefined.

Reply:
use req.body.category.id instead of req.body.id

Reply:
category.id is too much nesting for a single parameter.
Null/Undefined checks should be there